OWML.ModHelper.Interaction/ModInteraction.cs

```diff
@@ -1,36 +1,58 @@
 ﻿using OWML.Common;
-using OWML.ModHelper.Events;
 using System.Collections.Generic;
 using System.Linq;
+using UnityEngine;
 
 namespace OWML.ModHelper.Interaction
 {
     public class ModInteraction : IModInteraction
     {
-        private readonly List<ModBehaviour> _modList = new List<ModBehaviour>();
-        private readonly Dictionary<IModBehaviour, List<IModBehaviour>> _dependantDict = new Dictionary<IModBehaviour, List<IModBehaviour>>();
+        private readonly IList<ModBehaviour> _modList;
+        private Dictionary<ModBehaviour, List<ModBehaviour>> _dependantDict = new Dictionary<ModBehaviour, List<ModBehaviour>>();
+        private Dictionary<ModBehaviour, List<ModBehaviour>> _dependencyDict = new Dictionary<ModBehaviour, List<ModBehaviour>>();
 
-        public ModInteraction(List<ModBehaviour> list)
+        public ModInteraction(IList<ModBehaviour> list)
         {
             _modList = list;
 
+            RegenDicts();
+        }
+
+        private void RegenDicts()
+        {
+            _dependantDict = new Dictionary<ModBehaviour, List<ModBehaviour>>();
+            _dependencyDict = new Dictionary<ModBehaviour, List<ModBehaviour>>();
             foreach (var mod in _modList)
             {
-                var dependants = new List<IModBehaviour>();
+                var dependants = new List<ModBehaviour>();
+                var dependencies = new List<ModBehaviour>();
                 foreach (var dependency in _modList)
-                {
-                    if (dependency.ModHelper.Manifest.Dependencies != new string[] { } && dependency.ModHelper.Manifest.Dependencies.Contains(mod.ModHelper.Manifest.Name))
+                { 
+                    if (dependency.ModHelper.Manifest.Dependencies != new string[] { } && dependency.ModHelper.Manifest.Dependencies.Contains(mod.ModHelper.Manifest.UniqueName))
```

Comment:
`dependency.ModHelper.Manifest.Dependencies != new string[] { }` can be simplified to `dependency.ModHelper.Manifest.Dependencies.Any()` (.Any() is same as .Count == 0)

Reply:
shouldn't it be !dependency.ModHelper.Manifest.Dependencies.Any() ?
i'm reading that .Any() returns true if there are any elements in a sequence

Reply:
oh wait, nevermind

Reply:
done!